src/Httpful/Request.php

```diff
@@ -168,6 +176,23 @@ public function timeoutIn($seconds)
     }
 
     /**
+     * Specify a HTTP connection timeout
+     * @param float|int $timeout seconds to timeout the HTTP connection
+     * @return Request
+     * @throws Exception
+     */
+    public function setConnectionTimeout($connection_timeout)
+    {
+        if (!preg_match('/^\d+(\.\d+)?/', $connection_timeout)) {
```

Comment:
Feels like an odd way to do this. I think I would prefer that this actually checked if it was an int/float and not support strings.